### PR TITLE
fix(Styleguide): Add sectioning in a nice way

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Found a better native way to do this with props tables.

* Utilizes Storybook ArgsTable categories for system props.  Adding the correct groups through the addon enhancer.
* Ensures `theme` and `as` are categorized under a specific area.

Screen Shot 2020-12-06 at 1 02 05 PM

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.0.7-alpha.b8506f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fcd206a505f01a98f804aea--gamut-preview.netlify.app>https://5fcd206a505f01a98f804aea--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fcd206a505f01a98f804aea>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fce497aded7dd545df57964--gamut-preview.netlify.app>https://5fce497aded7dd545df57964--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fce497aded7dd545df57964>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.0.7-alpha.42ddbf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fce4ba27bde060170ae7a85--gamut-preview.netlify.app>https://5fce4ba27bde060170ae7a85--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fce4ba27bde060170ae7a85>Deploy Logs</a></p>

File: packages/styleguide/.storybook/addons/system/enhancers.ts [hunk 1]
```diff
@@ -29,54 +28,89 @@ const sortScale = (scale: string[]) => {
...
+          'Configures what element tag this component should present as',
+        table: {
+          ...arg.table,
+          category: 'base',
+        },
+        control: {
+          ...arg.control,
+          type: 'select',
+          options: map(
+            (val) => (typeof val === 'string' ? val.replace(/"/g, '') : val),
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/.storybook/addons/system/enhancers.ts:67]: I'm wondering if the code duplication around here is a sign that we should DRY it up...

File: packages/styleguide/.storybook/addons/system/enhancers.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
-import React from 'react';
-import { mapValues, update, isNumber, compose, get, set, map } from 'lodash/fp';
+import { mapValues, isNumber, get, map } from 'lodash/fp';
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/.storybook/addons/system/enhancers.ts:1]: 😂 , I forgot about these...

**Review by JoshuaKGoldberg — APPROVED** [review]
Cool!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/styleguide@21.0.7-alpha.02df42.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fce5c3869d69112cf53ebd1--gamut-preview.netlify.app>https://5fce5c3869d69112cf53ebd1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fce5c3869d69112cf53ebd1>Deploy Logs</a></p>

**Pull request merged**